Dl fix2

bbuzzanga opened the pull request:
work around in response to https://github.com/aria-tools/ARIA-tools/issues/229

Currently testing on track 121; any additional tests are welcome

bbuzzanga committed:
Limit file list given to ASF to 200 files/thread

bbuzzanga committed:
bug fixes for dl in chunks

dbekaert commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/aria-tools/ARIA-tools/pullRequest?prid=6271815&bid=20415860)

dbekaert commented on tools/bin/ariaDownload.py:
```diff
@@ -260,43 +257,55 @@ def make_nc_cookie(self):
+def _dl_helper(inp_dct):
     """ Helper function for parallel processing """
+    prod_dct = {'product_list': ','.join(inp_dct['files'])}
+    log.debug ('# of files: ', len(inp_dct['files']))
```
Codacy found an issue: [Too many arguments for logging format string](https://app.codacy.com/gh/aria-tools/ARIA-tools/pullRequest?prid=6271815)

dbekaert commented on tools/bin/ariaDownload.py:
```diff
@@ -260,43 +257,55 @@ def make_nc_cookie(self):
+    for chunk in chunks1:
+        chunks     = np.array_split(chunk, nt)
+        lst_dcts   = [vars(inps)]*nt  # Namespace->dctionary, repeat it
+        for i in range(len(chunks)): # put split up files to the objects for threads
```
Codacy found an issue: [Consider using enumerate instead of iterating with range and len](https://app.codacy.com/gh/aria-tools/ARIA-tools/pullRequest?prid=6271815)

dbekaert commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/bekaertdavid/ARIA-tools/pullRequest?prid=6271814&bid=20415859)

bbuzzanga committed:
address codacy

dbekaert commented:
tagging @rzinke to test this for his issue.

rzinke commented:
Hi @bbuzz31, thanks for looking into this. I tested for Track 41 and I am getting a new error.

CRITICAL: Downloader missed: S1-GUNW-A-R-041-tops-20141113_20141020-115903-35778N_33900N-PP-cf7d-v2_0_3, grab manually with date range
CRITICAL: Downloader missed: S1-GUNW-A-R-041-tops-20141113_20141020-115838-34285N_32405N-PP-1051-v2_0_1, grab manually with date range
CRITICAL: Downloader missed: S1-GUNW-A-R-041-tops-20141113_20141020-115838-34285N_32405N-PP-350c-v2_0_3, grab manually with date range
CRITICAL: Downloader missed: S1-GUNW-A-R-041-tops-20141113_20141020-115813-32793N_30910N-PP-a498-v2_0_1, grab manually with date range
CRITICAL: Downloader missed: S1-GUNW-A-R-041-tops-20141113_20141020-115813-32793N_30910N-PP-a4e3-v2_0_3, grab manually with date range
CRITICAL: Downloader missed: S1-GUNW-A-R-041-tops-20141113_20141020-115748-31300N_29248N-PP-6040-v2_0_3, grab manually with date range
CRITICAL: Downloader missed: S1-GUNW-A-R-041-tops-20141113_20141020-115721-29806N_27584N-PP-a54f-v2_0_3, grab manually with date range
Traceback (most recent call last):
  File "/u/sar-r0/rzinke/python/miniconda3/envs/ARIA-tools/bin/ariaDownload.py", line 321, in <module>
    Downloader(inps)()
  File "/u/sar-r0/rzinke/python/miniconda3/envs/ARIA-tools/bin/ariaDownload.py", line 102, in __call__
    os.remove(fileName)
NameError: name 'fileName' is not defined
(ARIA-tools) [rzinke@sar Track041]$ ls


Attached is my error.log file
[error.log](https://github.com/aria-tools/ARIA-tools/files/5320667/error.log)

bbuzzanga commented:
Thanks @rzinke I'm getting this also; still seems fine for splitting up the list manually though

bbuzzanga committed:
correct fileremove bug

sssangha committed:
Specified download of compatible GDAL version

sssangha committed:
Kick out shorter of the duplicate scenes

sssangha committed:
Update environment with gdal range

sssangha committed:
Update environment attempt 2

With the previous update to query within a defined range of gdal versions, the program defaults to a newer version outside of the range.

ehavazli committed:
Merge pull request #235 from sssangha/sss_dupbug

Improve handling of duplicate products

bbuzzanga committed:
Correctly split download files for parallel/ASF limits

Rewrite summary messages
Bug fixes

bbuzzanga committed:
remove deprecated code

bbuzzanga committed:
Merge branch 'dl_fix3' into dl_fix2

dbekaert commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 5
           
```


See the complete overview on [Codacy](https://app.codacy.com/gh/aria-tools/ARIA-tools/pullRequest?prid=6271815&bid=20415860)

dbekaert commented on tools/bin/ariaDownload.py:
```diff
@@ -260,45 +256,60 @@ def make_nc_cookie(self):
+    fpath    = os.path.abspath(os.path.join(inp_dct['wd'], fname))
+    with open(fpath, 'w') as f: f.write(script)
+    AD = __import__(os.path.splitext(fname)[0])
+    args, os.sys.argv = os.sys.argv, [] # gets around spurious messages
```
Codacy found an issue: [Unused variable 'args'](https://app.codacy.com/gh/aria-tools/ARIA-tools/pullRequest?prid=6271815)

dbekaert replied:
Codacy found an issue: [local variable 'args' is assigned to but never used (F841)](https://app.codacy.com/gh/aria-tools/ARIA-tools/pullRequest?prid=6271815)

dbekaert commented:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 4
           
```


See the complete overview on [Codacy](https://app.codacy.com/manual/bekaertdavid/ARIA-tools/pullRequest?prid=6271814&bid=20415859)

dbekaert commented on tools/bin/ariaDownload.py:
```diff
@@ -307,6 +318,32 @@ def check_cookie_is_logged_in(cj):
+
+
+    log.info ('Successes: %d files, %s bytes', len(succeed), tot_bytes)
+    [print ('\n\t-', sf['file'], f'{sf["size"]/1024**2:.2f}') for sf in succeed]
```
Codacy found an issue: [Expression "[print('\n\t-', sf['file'], f'{(sf['size']) / ((1024) ** (2))}') for sf in succeed]" is assigned to nothing](https://app.codacy.com/gh/aria-tools/ARIA-tools/pullRequest?prid=6271815)

dbekaert commented on tools/bin/ariaDownload.py:
```diff
@@ -307,6 +318,32 @@ def check_cookie_is_logged_in(cj):
+    [print ('\n\t-', sf['file'], f'{sf["size"]/1024**2:.2f}') for sf in succeed]
+
+    if skipped: log.info('Skipped: %d files', len(skipped))
+    [print ('\n\t-', sf) for sf in skipped]
```
Codacy found an issue: [Expression "[print('\n\t-', sf) for sf in skipped]" is assigned to nothing](https://app.codacy.com/gh/aria-tools/ARIA-tools/pullRequest?prid=6271815)

dbekaert commented on tools/bin/ariaDownload.py:
```diff
@@ -307,6 +318,32 @@ def check_cookie_is_logged_in(cj):
+    [print ('\n\t-', sf) for sf in skipped]
+
+    if failed: log.info('Failures: %d files', len(failed))
+    [print ('\n\t-', sf) for sf in failed]
```
Codacy found an issue: [Expression "[print('\n\t-', sf) for sf in failed]" is assigned to nothing](https://app.codacy.com/gh/aria-tools/ARIA-tools/pullRequest?prid=6271815)
